### DISCUSSION TITLE
TOC Not Rendering After R Code Chunk in Quarto

### DISCUSSION THREAD

**Discussion by ArtDowdy** [quarto-dev/quarto-cli#11287] [Q&A]
### Description

I'm encountering a problem with Quarto in which the Table of Contents (TOC) stops rendering correctly after a specific R code chunk. Although headings prior to the code chunk are displayed in the TOC, all headings following the code chunk are missing from the TOC, even though they appear in the main document content.

YAML Configuration
Here’s my YAML setup:

yaml
---
title: "A Scoping Review of MCMC Applications in Single-Case Experimental Design Research"
format:
  html:
    theme: cosmo
    toc: true
    toc-location: left
    toc-depth: 3
    code-fold: true  
    code-tools: true
    css: styles.css
    self-contained: true
    math: katex
    highlight-style: tango
---
Problematic Section in Document
The issue begins after the following heading and code chunk:

## 3D MCMC Simulation Plot
The 3D plot below provides a visualization of the Markov Chain Monte Carlo (MCMC) process over time, specifically across two parameters, $\theta_1$ and $\theta_2$. Each point in the plot represents one iteration of the MCMC process, where new values for these parameters are proposed and either accepted or rejected according to the Metropolis-Hastings algorithm.
...
r
<div style="width: 100%; height: 600px;">
```{r}
library(ggplot2)
library(plotly)

run_mcmc_3d <- function(n_iter, target_mean = c(0, 0), target_sd = c(1, 1), proposal_sd = c(0.5, 0.5)) {
  # MCMC Simulation Code
  ...
}
n_iter <- 1000
mcmc_result <- run_mcmc_3d(n_iter)

interactive_plot_3d <- plot_ly(...)
interactive_plot_3d
</div> ```

TOC Behavior Observed
1. Before this code chunk: All headings display as expected in the TOC.
2. After this code chunk: None of the following headings appear in the TOC, even though they are present in the document content.

Troubleshooting Steps Taken
1. Adjusted YAML Settings: I tried different toc-depth levels and removed optional settings (like math: katex and highlight-style) to simplify the YAML. The problem persisted.
2. Relocated Code Chunk: I attempted to move this code chunk further down, after additional text and headings, but the TOC still stopped displaying headings following this chunk.
3. Removed css: styles.css: I temporarily removed the custom CSS to rule out interference with TOC rendering, but this did not resolve the issue.
4. Checked include: false: I ensured that #| include: false was not present for this chunk, as that can suppress output and affect TOC display.

Expected vs. Actual Output
- Expected: A TOC that includes all level 2 and 3 headings throughout the document.
- Actual: The TOC stops displaying any headings after the "3D MCMC Simulation Plot" section.


Could certain code chunks (e.g., complex or interactive plots) disrupt TOC rendering in Quarto?
Are there specific settings or workarounds to prevent code chunks from interfering with TOC generation?
Any guidance or suggestions to help resolve this TOC issue would be greatly appreciated!

**Comment by mcanouil** [comment]
Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.
See <https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us>.

Please remove everything unnecessary from your code example. I highly doubt we need all those options.
Please do make sure your code is reproducible. If we can't use your code, it's unlikely we can help.

`self-contained` was superseded more than two years ago (since Quarto 1.1 or 1.2) and is no longer documented. Please update to `embed-resources`

**Reply by ArtDowdy** [reply]
Sure - thank you. ## Research Question is the first heading that doesn't show along with all below that heading.

## 3D MCMC Simulation Plot

The 3D MCMC simulation plot thus offers a valuable visual tool for assessing the chain’s behavior and convergence properties. Stable clustering of the points suggests successful convergence, providing confidence that the MCMC algorithm has effectively sampled from the high-probability regions of the target distribution. On the other hand, an absence of stabilization may prompt adjustments to the MCMC parameters or proposal distribution to improve convergence.

<div style="width: 100%; height: 600px;">

````
library(ggplot2)
library(plotly)

run_mcmc_3d <- function(n_iter, target_mean = c(0, 0), target_sd = c(1, 1), proposal_sd = c(0.5, 0.5)) {
  theta <- matrix(0, ncol = 2, nrow = n_iter)
  theta[1, ] <- rnorm(2, mean = target_mean, sd = target_sd)
  acceptance_rate <- 0
  
  target_density <- function(x) {
    dnorm(x[1], mean = target_mean[1], sd = target_sd[1]) * dnorm(x[2], mean = target_mean[2], sd = target_sd[2])
  }
  
  for (i in 2:n_iter) {
    theta_proposed <- rnorm(2, mean = theta[i-1, ], sd = proposal_sd)
    accept_prob <- min(1, target_density(theta_proposed) / target_density(theta[i-1, ]))
    
    if (runif(1) < accept_prob) {
      theta[i, ] <- theta_proposed
      acceptance_rate <- acceptance_rate + 1
    } else {
      theta[i, ] <- theta[i-1, ]
    }
  }
  
  return(list(chain = theta, acceptance_rate = acceptance_rate / n_iter))
}

n_iter <- 1000
mcmc_result <- run_mcmc_3d(n_iter)

mcmc_data <- data.frame(
  Iteration = 1:n_iter,
  Theta1 = mcmc_result$chain[, 1],
  Theta2 = mcmc_result$chain[, 2]
)

interactive_plot_3d <- plot_ly(
  mcmc_data, x = ~Iteration, y = ~Theta1, z = ~Theta2, 
  type = 'scatter3d', mode = 'lines+markers', 
  marker = list(size = 3, color = ~Iteration, colorscale = 'Viridis'),
  line = list(color = 'blue')
)

interactive_plot_3d <- interactive_plot_3d %>%
  layout(
    title = '3D MCMC Simulation: Trace Plot',
    scene = list(
      xaxis = list(title = 'Iteration'),
      yaxis = list(title = 'Theta1'),
      zaxis = list(title = 'Theta2')
    ),
    autosize = TRUE,
    width = '100%',
    height = '100%'
  )

interactive_plot_3d

````
## Research Question

How has MCMC been applied in single-case experimental designs (SCED)?

**Reply by mcanouil** [reply]
That's not better, please edit your original post and follow the guidelines provided. Thanks you.
